crates/re_log_types/src/lib.rs

```diff
@@ -130,11 +125,15 @@ impl std::fmt::Display for RecordingId {
     }
 }
 
-impl std::str::FromStr for RecordingId {
-    type Err = <uuid::Uuid as std::str::FromStr>::Err;
+impl From<&str> for RecordingId {
+    fn from(s: &str) -> Self {
+        Self(Arc::new(s.to_owned()))
+    }
+}
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        s.parse().map(Self)
+impl From<String> for RecordingId {
+    fn from(s: String) -> Self {
+        Self(Arc::new(s))
     }
 }
```

Comment:
As noted in the other PR: you can also add `impl`s for `AsRef`, `Borrow` and `Deref`. See `crates/re_log_types/src/component_types/label.rs`